docs/architecture/adr-018-social-upgrades.md

```diff
@@ -0,0 +1,67 @@
+One of celestia's core principles is social consensus, this is why the standard cosmos-sdk upgrade mechanism that relies on token voting has been removed. We still need a mechanism to upgrade after social consensus has been reached, and that is what this document aims to clarify.
+
+### Rolling vs Stopping Upgrades
+
+A rolling upgrade involves nodes upgrading to the binary ahead of time, and that new binary will automatically switch to the new consensus logic at a provided height. Stopping upgrades require all nodes in the network to halt at a provided height, and collectively switch binaries. All upgrades can occur in a rolling or stopping fashion, but doing so in a rolling fashion requires more work. Fortunately, since we're already supporting single binary syncs, the vast majority of upgrades will be able to roll with very little additional changes.
+
+There are however still a very small percentage of changes that require significantly more work to become rolling. It can still be done, it simply requires more work per each of those types of upgrades.
+
+### Balancing Hardfork and Halting Risk
+
+One of the main difficulties of social upgrades when using tendermint consensus is finding a balance between risking a halt and ending up in a situation where the community must hardfork. If social consensus is reached, but validators do not incorporate the upgrade, then a hardfork must be performed. This involves changing the chain-id and the validator set, which would force the governance of connected IBC chains to recognize the changes in order to preserve celestia token holders that have bridged to one of those chains.
+
+One mechanism that has been proposed is to add some halt height (aka difficulty bomb) for light clients and consensus nodes. This halt height could be determined before the upgrade binary is released, or it could be incorporated to the upgraded binary. The important feature of such mechanisms is to set a deadline for validators to upgrade. If a solution cannot be agreed upon by all parties offchain by that point, then a fork will be created by the community.
+
+### Deciding And Relaying Upgrade Height
+
+One of the main decisions that needs to be made is how to decide and convey the upgrade height to the application. Each has trade offs on the halting vs hardfork risk.
+
+#### Hardcoded
+
+The upgrade height can be hardcoded into the binary. When that upgrade height is reached, the relevant logic is routed appropriately. This approach is simple but not flexible. We run many different testnets, including countless ephemeral ones, and each will have a different upgrade height. Not only do all the upgrade heights for each network have to be handled, but it requires a new major release to change. This means that if for some reason social consensus decides to postpone the upgrade, then a new release must be created. The main risk is that this approach is prone to errors that could halt the network if social consensus is not reached.
```

Comment:
> including countless ephemeral ones, and each will have a different upgrade height

[question] can the hard-coded upgrade height be applicable on a per chain-id & block height basis?

Reply:
Disregard, this option is explored in `#### Configured`